tests/components/knx/test_climate.py

```diff
@@ -0,0 +1,237 @@
+    # read target temperature state
+    await knx.assert_read("1/2/5")
+
+    # turn hvac off
+    await hass.services.async_call(
+        "climate",
+        "set_hvac_mode",
+        {"entity_id": "climate.test", "hvac_mode": HVAC_MODE_OFF},
+        blocking=True,
+    )
+    await knx.assert_write("1/2/8", False)
+
+    # turn hvac on
+    await hass.services.async_call(
+        "climate",
+        "set_hvac_mode",
+        {"entity_id": "climate.test", "hvac_mode": HVAC_MODE_HEAT},
+        blocking=True,
+    )
+    await knx.assert_write("1/2/8", True)
+    await knx.assert_write("1/2/6", (0x01,))
+
+
+async def test_climate_preset_mode(hass: HomeAssistant, knx: KNXTestKit):
+    """Test KNX climate preset mode."""
+    events = async_capture_events(hass, "state_changed")
+    await knx.setup_integration(
+        {
+            ClimateSchema.PLATFORM: {
+                CONF_NAME: "test",
+                ClimateSchema.CONF_TEMPERATURE_ADDRESS: "1/2/3",
+                ClimateSchema.CONF_TARGET_TEMPERATURE_ADDRESS: "1/2/4",
+                ClimateSchema.CONF_TARGET_TEMPERATURE_STATE_ADDRESS: "1/2/5",
+                ClimateSchema.CONF_OPERATION_MODE_ADDRESS: "1/2/6",
+                ClimateSchema.CONF_OPERATION_MODE_STATE_ADDRESS: "1/2/7",
+            }
+        }
+    )
+    assert len(hass.states.async_all()) == 1
+    assert len(events) == 1
+    events.pop()
+
+    await hass.async_block_till_done()
+    # read states state updater
+    await knx.assert_read("1/2/7")
+    await knx.assert_read("1/2/3")
+    # StateUpdater initialize state
+    await knx.receive_response("1/2/7", True)
+    await knx.receive_response("1/2/3", (0x01,))
+    # StateUpdater semaphore allows 2 concurrent requests
+    # read target temperature state
+    await knx.assert_read("1/2/5")
+
+    # set preset mode
+    await hass.services.async_call(
+        "climate",
+        "set_preset_mode",
+        {"entity_id": "climate.test", "preset_mode": PRESET_ECO},
+        blocking=True,
+    )
+    await knx.assert_write("1/2/6", (0x04,))
+    assert len(events) == 1
+    events.pop()
+
+    # set preset mode
+    await hass.services.async_call(
+        "climate",
+        "set_preset_mode",
+        {"entity_id": "climate.test", "preset_mode": PRESET_SLEEP},
+        blocking=True,
+    )
+    await knx.assert_write("1/2/6", (0x03,))
+    assert len(events) == 1
+    events.pop()
+
+    assert len(knx.xknx.devices) == 2
+    assert len(knx.xknx.devices[0].device_updated_cbs) == 2
+    assert len(knx.xknx.devices[1].device_updated_cbs) == 2
+    # test removing also removes hooks
+    er = entity_registry.async_get(hass)
+    er.async_remove("climate.test")
+    await hass.async_block_till_done()
+
+    assert len(knx.xknx.devices) == 2
+    assert len(knx.xknx.devices[0].device_updated_cbs) == 1
+    assert len(knx.xknx.devices[1].device_updated_cbs) == 1
+
+
+async def test_update_entity(hass: HomeAssistant, knx: KNXTestKit, mock_ha):
```

Comment:
```suggestion
async def test_update_entity(hass: HomeAssistant, knx: KNXTestKit):
```
unused fixture

Reply:
It is used. Otherwise the servicecall fails.